GVFS/GVFS.UnitTests/Common/GVFSEnlistmentHealthTests.cs

```diff
@@ -101,59 +105,68 @@ public void VariedDirectoryFormatting()
+                .AddModifiedPathFiles("B/1.js", $"B{this.sep}2.js", "/B/1.js", $"{this.sep}B{this.sep}1.js")
+                .Build();
 
-            this.enlistmentHealthData = this.GenerateStatistics(string.Empty);
+            this.enlistmentHealthData = this.GenerateStatistics(pathData, string.Empty);
 
-            this.enlistmentHealthData.PlaceholderCount.ShouldEqual(4);
-            this.enlistmentHealthData.ModifiedPathsCount.ShouldEqual(4);
-            this.enlistmentHealthData.GitTrackedItemsCount.ShouldEqual(6);
+            this.enlistmentHealthData.PlaceholderCount.ShouldEqual(pathData.PlaceholderFilePaths.Count);
+            this.enlistmentHealthData.ModifiedPathsCount.ShouldEqual(pathData.ModifiedFilePaths.Count);
+            this.enlistmentHealthData.GitTrackedItemsCount.ShouldEqual(pathData.GitFilePaths.Count);
             this.enlistmentHealthData.DirectoryHydrationLevels.Count.ShouldEqual(2);
-            this.enlistmentHealthData.PlaceholderPercentage.ShouldEqual(2.0m / 3.0m);
-            this.enlistmentHealthData.ModifiedPathsPercentage.ShouldEqual(2.0m / 3.0m);
+            this.enlistmentHealthData.PlaceholderPercentage.ShouldEqual((decimal)pathData.PlaceholderFilePaths.Count / (decimal)pathData.GitFilePaths.Count);
+            this.enlistmentHealthData.ModifiedPathsPercentage.ShouldEqual((decimal)pathData.ModifiedFilePaths.Count / (decimal)pathData.GitFilePaths.Count);
             this.enlistmentHealthData.DirectoryHydrationLevels[0].Value.ShouldEqual(4.0m / 3.0m);
             this.enlistmentHealthData.DirectoryHydrationLevels[1].Value.ShouldEqual(4.0m / 3.0m);
         }
 
         [TestCase]
         public void FilterByDirectory()
         {
-            this.fauxGitFiles.AddRange(new string[] { "A/1.js", "A/2.js", "A/3.js", "B/1.js", "B/2.js", "B/3.js" });
-            this.fauxPlaceholderFiles.AddRange(new string[] { "A/1.js", $"A{this.sep}2.js", "/A/1.js", $"{this.sep}A{this.sep}1.js" });
-            this.fauxModifiedPathsFiles.AddRange(new string[] { "B/1.js", $"B{this.sep}2.js", "/B/1.js", $"{this.sep}B{this.sep}1.js" });
+            GVFSEnlistmentHealthCalculator.GVFSEnlistmentPathData pathData = new PathDataBuilder()
+                .AddGitFiles("A/1.js", "A/2.js", "A/3.js", "B/1.js", "B/2.js", "B/3.js")
+                .AddPlaceholderFiles("A/1.js", $"A{this.sep}2.js", "/A/1.js", $"{this.sep}A{this.sep}1.js")
+                .AddModifiedPathFiles("B/1.js", $"B{this.sep}2.js", "/B/1.js", $"{this.sep}B{this.sep}1.js")
+                .Build();
 
-            this.enlistmentHealthData = this.GenerateStatistics("A/");
+            this.enlistmentHealthData = this.GenerateStatistics(pathData, "A/");
 
-            this.enlistmentHealthData.GitTrackedItemsCount.ShouldEqual(3);
-            this.enlistmentHealthData.PlaceholderCount.ShouldEqual(4);
+            this.enlistmentHealthData.GitTrackedItemsCount.ShouldEqual(pathData.GitFilePaths.Count / 2);

```

Comment:
instead of `pathData.GitFilePaths.Count / 2`, could we do something like `expectedCountofFilesInDirectoryA`, or possibly do something like:

```
string [] filesPathsDirectoryA = new string [ {"A/1.js", "A/2.js", ....};
string [] dilesPathsDirectoryB = new string [ {"B/1.js", "/B/2.js", ....};

new PathDataBuild().AddGitFiles(filePathsDirectoryA).AddGitFiles(filePathsDirectoryB)....
this.enlistmentHealthData.GitTrackedItemsCount.ShouldEqual(filesPathsDirectoryA.Count);
```